test-poc/framework/README.md

```diff
@@ -0,0 +1,336 @@
+# Introduction
+
+The Keycloak JUnit 5 test framework makes it easy to write tests for Keycloak and extensions. Behind the scenes the
+framework handles the lifecycle of Keycloak, the database, and any injected resources such as realms and clients.
+
+Tests simply declare what they want, including specific configuration, and the framework takes care of the rest. 
+
+
+# Writing tests
+
+An example is better than a lot of works, so here is a very basic test:
+
+```java
+@KeycloakIntegrationTest
+public class BasicTest {
+
+    @InjectRealm
+    ManagedRealm realm;
+
+    @Test
+    public void test() {
+        Assertions.assertEquals("default", realm.getName());
+        Assertions.assertEquals(1, realm.admin().users().list().size());
+    }
+
+}
+```
+
+## Resource lifecycle
+
+Managed resources can have the following life-cycles:
+
+* Global - Used across multiple test classes
+* Class - Used for a single test class, and re-used for multiple test methods
+* Method - Use if a single test method
+
+The framework handles the lifecycle accordingly to how it is configured in the annotation, or the default lifecycle
+for a given resource.
+
+For example the default lifecycle for a realm is Class, but it can be changed through the annotation:
+
+```java
+@InjectRealm(lifecycle = LifeCycle.METHOD)
+ManagedRealm realm;
+
+@Test
+public void test() {
+    realm.admin().users().create(...);
+}
+
+@Test
+public void test() {
+    Assertions.assertEquals(0, realm.admin().users().list().size());
+}
+```
+
+When the lifecycle is set to Method the realm is automatically destroyed and re-created for each test method, as seen in 
+the above example where one test method adds a user to the realm, but the user is not present in the next test.
+
+The general recommendation is to use the Class lifecycle for realms, clients, and users. Making sure that individual test
+methods leave the resource in a way that can be re-used. Realms for example with global lifecycle can be harder to
+maintain as individual test classes can break other tests, but at the same time using global resources can be useful 
+as it will be less performant.
```

Comment:
I am confused by the wording. Wouldn't it be more performant?

Reply:
Typo (of sorts) updated